src/modules/generateRandomNumber.js

```diff
@@ -6,8 +6,17 @@
  *
  * @return {number} A random 4-digit number
  */
+
 function generateRandomNumber() {
-  /* Write your code here */
+  const uniqRandomNumbers = new Set();
+
+  while (uniqRandomNumbers.size !== 4) {
+    const randomNumber = Math.floor(1 + Math.random() * 9);
```

Comment:
The current logic for generating random numbers only produces digits from 1 to 9. This ensures no leading zero, but the logic for ensuring exactly 4 unique digits is flawed. Consider modifying the logic to ensure the first digit is non-zero and the remaining digits are unique, potentially by generating the first digit separately and then filling the rest.